src/loki/search.cc

```diff
@@ -476,11 +474,14 @@ struct bin_handler_t {
 
     // if the inbound reach is not 0 and the outbound reach is not 0 and the opposing edge is not
     // filtered then the reaches of both edges are the same
-    const DirectedEdge* opp_edge = nullptr;
-    if (reach.outbound > 0 && reach.inbound > 0 && (opp_edge = reader.GetOpposingEdge(edge, tile)) &&
-        edge_filter(opp_edge) > 0.f)
-      directed_reaches[opp_edge] = reach;
-
+    if (reach.outbound > 0 && reach.inbound > 0) {
+      const DirectedEdge* opp_edge = nullptr;
+      const auto opp_edgeid = reader.GetOpposingEdgeId(edge_id, tile);
+      if (opp_edgeid && (opp_edge = tile->directededge(opp_edgeid)) &&
```

Comment:
i think id prefer not changing this to be a doubly nested `if`. it may have no performance impact or it may, we'll have to do extra testing to find out and i am not sure it really improves readibility. i guess in this case i would have modified the helper functions in graph reader to get both the edge and the edge id? maybe that would be more useful?